pkg/codegen/dotnet/gen_program.go

```diff
@@ -157,7 +158,17 @@ func GenerateProgram(program *pcl.Program) (map[string][]byte, hcl.Diagnostics,
 	return GenerateProgramWithOptions(program, defaultOptions)
 }
 
-func GenerateProject(directory string, project workspace.Project, program *pcl.Program) error {
+func localDotnet(localProjects map[string]map[string]string, pkg string) (string, bool) {
+	local, ok := localProjects[pkg]
+	if !ok {
+		return "", false
+	}
+	dotnet, ok := local["dotnet"]
+	return dotnet, ok
+}
+
+func GenerateProject(directory string, project workspace.Project, program *pcl.Program,
+	localProjects map[string]map[string]string) error {
```

Comment:
```suggestion
	localProjects map[string]string) error {
```
Resolve to the "dotnet" map in the caller of this method. There's no point passing a map here that we're only going to look at the "dotnet" key in.